cmd/ttn-lw-cli/commands/use.go

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	configFileName = ".ttn-lw-cli.yml"
+	useCommand     = &cobra.Command{
+		Use:               "use",
+		Short:             "Use",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			insecure, _ := cmd.Flags().GetBool("insecure")
+			ca, _ := cmd.Flags().GetString("ca")
```

Comment:
1. No need to do any `ca` stuff if `insecure == true`
2. We could also consider adding another flag to load a custom CA by connecting to the deployment:

```go
conn, err := tls.Dial("tcp", host+":8884", &tls.Config{InsecureSkipVerify: true})
if err != nil {
	return err
}
defer conn.Close()
f, err := os.OpenFile(filepath.Join(dir, "ca.pem"))
defer func() {
	if closeErr := f.Close(); err == nil && closeErr != nil {
		err = closeErr
	}
}()
for _, cert := range conn.ConnectionState().PeerCertificates {
	if !cert.IsCA {
		continue
	}
	pem.Encode(f, &pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
}
```

Reply:
No CA stuff is done if `insecure == true`. Does this not suffer by the same security issues discussed in #2156?

Reply:
So, one thing I do not really like about this is that we have to make sure to sanitize `host` before trying to connect to port 8884, which is generally not an easy problem in itself.